src/edu/ucsb/nceas/metacat/dataone/MNodeService.java

```diff
@@ -2209,17 +2212,23 @@ public Identifier publish(Session session, Identifier originalIdentifier) throws
 				oreSysMeta.setObsoletedBy(null);
 				oreSysMeta.setSize(BigInteger.valueOf(resourceMapString.getBytes("UTF-8").length));
 				oreSysMeta.setChecksum(ChecksumUtil.checksum(resourceMapString.getBytes("UTF-8"), oreSysMeta.getChecksum().getAlgorithm()));
+				oreSysMeta.setFileName("resourceMap_" + newOreIdentifier.getValue() + ".rdf.xml");
 				
 				// ensure ORE is publicly readable
-				oreSysMeta = makePublicIfNot(oreSysMeta, potentialOreIdentifier);
-				
+                oreSysMeta = makePublicIfNot(oreSysMeta, potentialOreIdentifier);
```

Comment:
The new system metadata of the new resource map object was copied and modified from the old resource map system metadata. Since the old resource map is not necessary to be public readable, we do a check here.